WordPress/src/test/java/org/wordpress/android/ui/posts/PostListMainViewModelTest.kt

```diff
@@ -186,4 +195,50 @@ class PostListMainViewModelTest : BaseUnitTest() {
         // assert
         verify(editPostRepository, times(0)).loadPostByLocalPostId(any())
     }
+
+    @Test
+    fun `if PostStatus is DRAFT and Publish Date is in the past then call publish immediately use case`() {
+        // arrange
+        val editPostRepository: EditPostRepository = mock()
+        whenever(editPostRepository.status).thenReturn(DRAFT)
+        whenever(editPostRepository.dateCreated).thenReturn("")
+        whenever(postUtilsWrapper.isPublishDateInThePast(any())).thenReturn(true)
+
+        // act
+        viewModel.start(site, PostListRemotePreviewState.NONE, currentBottomSheetPostId, editPostRepository)
+        viewModel.showPrepublishingBottomSheet(mock())
+
+        // assert
+        verify(publishPostImmediatelyUseCase, times(1)).updatePostToPublishImmediately(any(), any())
+    }
+
+    @Test
+    fun `if PostStatus is DRAFT and Publish Date is is in the future then don't call publish immediately use case`() {
```

Comment:
Is there a scenario in which the user clicks on "Publish" but the date is in the future? Do we really want to show the future date but publish the post immediately?

Reply:
No, we don't! b51fd05